### PR TITLE
Fix NameError in idpicker.py on Python 3.

### LINKED ISSUE #7: NameError: name 'unicode' is not defined under Python 3
Porting zope.app.authentication to Python 3 reveals this error in idpicker.py:

```
     File "//site-packages/zope/app/container/browser/adding.py", line 75, in add
        name = chooser.chooseName('', content)
      File "//site-packages/zope/pluggableauth/plugins/idpicker.py", line 56, in chooseName
        name = unicode(name)
    NameError: name 'unicode' is not defined
```

### PR DESCRIPTION
**Pull request by jamadden**
Fixes #7.

There were tests for this, they just weren't being run, so make sure to run them.

Also a number of small updates:

- Run *all* the doctests.
- Enable test coverage reporting.
- Add support for Python 3.6 and explicitly list PyPy in the classifiers too.
- Modernize tox.ini
- Use pip on Travis and enable Travis caching.
- Run all doctests with consistent flags and checker
- Use renormalizing.IGNORE_EXCEPTION_MODULE_IN_PYTHON2 instead of
  manually normalizing one exception
- Rename .txt to .rst and include them in the published docs. Fix
  headers to be consistent to enable this.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -28,21 +28,28 @@ def read(*rnames):
...
 
+tests_require = [
+    'zope.testing',
+    'zope.testrunner',
+]
+
 setup(
     name='zope.pluggableauth',
-    version='2.1.1.dev0',
+    version='2.2.1.dev0',
```
Comment (tseaver) [inline setup.py]: Should be `2.2.0.dev0`.

File: src/zope/pluggableauth/README.rst [hunk 2]
```diff
@@ -1,5 +1,6 @@
-Pluggable-Authentication Utility
-================================
+==================================
+ Pluggable-Authentication Utility
+==================================
```
Comment (tseaver) [inline src/zope/pluggableauth/README.rst]: What is the rationale for changing the RST headers in this PR?

File: src/zope/pluggableauth/plugins/idpicker.py [hunk 3]
```diff
@@ -53,11 +59,11 @@ class IdPicker(NameChooser):
     """
     def chooseName(self, name, object):
         i = 0
-        name = unicode(name)
+        name = text_type(name)
         orig = name
         while (not name) or (name in self.context):
             i += 1
-            name = orig+str(i)
+            name = orig + str(i)
```
Comment (tseaver) [inline src/zope/pluggableauth/plugins/idpicker.py]: How is this not going to break where `text_type` is not the same as `str`?

File: src/zope/pluggableauth/README.rst (hunk 2 above)
Comment (jamadden) [inline src/zope/pluggableauth/README.rst]: All the files need to be consistent with each other in order to be combined.

File: src/zope/pluggableauth/plugins/idpicker.py (hunk 3 above)
Comment (jamadden) [inline src/zope/pluggableauth/plugins/idpicker.py]: It's not going to break mostly through luck 😄 

On Python 3, `text_type` is `str` and `str` + `str` works out as expected.
```pycon
>>> u'abc' + str(123)
'abc123'
```

On Python 2, `text_type` is `unicode`. `unicode` + `str` does unicode-promotion on the `str` argument; the `str` of an int is only going to have ASCII characters and so will always be promotable to unicode.
```pycon
>>> u'abc' + str(123)
u'abc123'
```

FWIW, `zope.container.contained.NameChooser` (from which this class derives) also makes the exact same assumptions in its `chooseName` method, freely combining `unicode` and `str`:
```python
        # convert to unicode and remove characters that checkName does not allow
        if isinstance(name, bytes):
            name = name.decode()
        if not isinstance(name, unicode):
            try:
                name = unicode(name)
            except:
                name = u''
        name = name.replace('/', '-').lstrip('+@')

        ...
        # for an existing name, append a number.
        # We should keep client's os.path.extsep (not ours), we assume it's '.'
        dot = name.rfind('.')
        if dot >= 0:
            suffix = name[dot:]
            name = name[:dot]
        else:
            suffix = ''

        n = name + suffix
        i = 1
        while n in container:
            i += 1
            n = name + u'-' + str(i) + suffix
```

File: setup.py (hunk 1 above)
Comment (jamadden) [inline setup.py]: Whoops. Thanks.

File: src/zope/pluggableauth/plugins/idpicker.py (hunk 3 above)
Comment (mgedmin) [inline src/zope/pluggableauth/plugins/idpicker.py]: "where `text_type` is not `str`" == Python 2, yes?
```
$ python2
>>> u'\N{SNOWMAN}-' + str(1)
u'\u2603-1'
```

str(number) gives pure-ASCII strings which can be automatically promoted to unicode on Python 2.

**Comment by mgedmin** [comment]
@jamadden: you now have PyPI rights for zope.pluggableauth

**Comment by jamadden** [comment]
Thanks for the reviews!

**Pull request merged**